lab-integration/src/main.ts

```diff
@@ -0,0 +1,85 @@
+import { WatchEventType, promises as fs, watch } from "fs";
+import * as lockfile from "proper-lockfile";
+import path from "path";
+import { queue } from "async";
+import puppeteer from "puppeteer";
+
+import { DateTime } from "luxon";
+import { TestsResult } from "./types";
+
+const filePath = path.join(process.cwd(), "../test-results.json");
+
+const machines = new Map<string, string>([
+  ["5534", "1 - Prensa PRE-01"],
+  ["5533", "2 - Prensa PRE-02"],
+  ["5532", "3 - Prensa PRE-03"],
+]);
+
+(async () => {
+  console.info("SCRIPT IS RUNNING...");
+
+  let lastTestResultDate = DateTime.now();
+
+  const browser = await puppeteer.launch({ headless: false, slowMo: 100 });
+  const page = await browser.newPage();
+  await page.setViewport({ width: 720, height: 720 });
+  await page.goto("http://localhost:3000/submission");
+
+  const queueTask = queue(
+    async (task: { eventType: WatchEventType }, callback) => {
+      try {
+        await lockfile.lock(filePath);
+        const stringifiedDataset = await fs.readFile(filePath, "utf8");
+        const parsedDataset = JSON.parse(stringifiedDataset) as TestsResult[];
+        const dataset = parsedDataset.map((testResult) => {
+          const Data = DateTime.fromFormat(
+            testResult.Data,
+            "dd/LL/yyyy, HH:mm:ss",
+            { locale: "pt-BR" },
+          );
+
+          return {
+            ...testResult,
+            Data: Data.isValid ? Data : DateTime.now(),
+          };
+        });
+
+        if (task.eventType === "change") {
+          const newTestsResults = dataset.filter(
+            (testResult) =>
+              testResult.Data.diff(lastTestResultDate).milliseconds > 0,
+          );
+          if (dataset.length > 0) {
+            lastTestResultDate = dataset.sort(
+              (a, b) => b.Data.diff(a.Data).milliseconds,
+            )[0].Data as unknown as DateTime<true>;
```

Comment:
Do we need `as unknown as DateTime<true>`? Because we already map the `parsedDataset` above.